presto-hive-common/src/main/java/org/apache/hadoop/fs/HadoopExtendedFileSystem.java

```diff
@@ -125,15 +125,15 @@ public Path resolvePath(final Path p)
     public FSDataInputStream open(Path f, int bufferSize)
             throws IOException
     {
-        return fs.open(f, bufferSize);
+        return new InputStreamWrapper(getRawFileSystem().open(f, bufferSize), this);
```

Comment:
These code might be duplicated with `PrestoFilterFileSystemWrapper`. I think we might be able to reuse the code by updating `PrestoExtendedFileSystemCache` as

```java
    protected FileSystem createPrestoFileSystemWrapper(FileSystem original)
    {
        return new HadoopExtendedFileSystem(super.createPrestoFileSystemWrapper(original));
    }
```

Reply:
@7c00  Thanks for you review, Your suggestion seems more reasonable, I will update it.